sklearn/neighbors/_classification.py

```diff
@@ -278,6 +295,38 @@ def predict_proba(self, X):
             by lexicographic order.
         """
         if self.weights == "uniform":
+            if (
+                self.algorithm == "brute"
+                and PairwiseDistancesArgKminLabels.is_usable_for(
+                    X, self._fit_X, self.metric
+                )
+            ):
+                if self.outputs_2d_:
+                    probabilities = []
+                    for k in range(self._y.shape[1]):
+                        probabilities.append(
+                            PairwiseDistancesArgKminLabels.compute(
+                                X,
+                                self._fit_X,
+                                k=self.n_neighbors,
+                                weights=self.weights,
+                                labels=self._y[:, k],
+                                metric=self.metric,
+                                metric_kwargs=self.metric_params,
+                            )
+                        )
```

Comment:
I think it is better to have this loop on labels be done in `weighted_histogram_mode`: currently this is doing `self._y.shape[1]` times as many computations as needed.